### PR TITLE
Fix test for `disk_list()` to be OS-agnostic and test the major device number filter

### PR DESCRIPTION
**Pull request by ashwin9390**
Fix the test of `disk_list()` to be host/OS-independent and test also the major device number filter in it:

- the 1st commit only prepares the code for the new test and the tested return value does not change in GitHub CI.
- the 2nd commit then updates the test to be a proper test, checking the branches in the function to perform as expected (e.g. devices with major device id >= 245 are filtered out)

In short, the commits are:

1. Update xen-bugtool.disk_list() to use the existing PROC_PARTITIONS variable (and also use `with open()`)
2. Update the test case to supply much better test data showing the filtering that the function is implemented to do to work.

### CODE REVIEW

**Review by bernhardkaindl — APPROVED** [review]
The code is from me, it fixes `pytest` to pass in Ashwin's XenDaaS development VM.

It tests only a tiny function that gets the non-xvda and non-dm disks.

**Review by GeraldEV — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
The update of the PR now cleans-up the code and the test-case.

It is split into 2 commits which are pushed one by one to let GitHub CI pass and show by the checkmark that:

- the 1st commit only changes the code and the tested return value does not change in GitHub CI.

- the 2nd commit then updates the test to be a proper test, checking the branches in the function to perform as expected (e.g. devices with major device id >= 245 are filtered out)

In short, the commits are:

1. Update xen-bugtool.disk_list() to use the existing PROC_PARTITIONS variable (and also use `with open()`)
2. Update the test case to supply much better test data showing the filtering that the function is implemented to do to work.

**Review by GeraldEV — APPROVED** [review]
